plugins/org.eclipse.n4js/src/org/eclipse/n4js/xtext/workspace/ProjectConfigSnapshot.java

```diff
@@ -10,57 +10,64 @@
  */
 package org.eclipse.n4js.xtext.workspace;
 
-import java.util.stream.Collectors;
-
 import org.eclipse.emf.common.util.URI;
+import org.eclipse.xtext.workspace.IProjectConfig;
 
-import com.google.common.collect.ImmutableCollection;
-import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableSet;
 
 /**
- * Default implementation of an {@link IProjectConfigSnapshot}.
+ * An immutable equivalent to {@link IProjectConfig}.
  */
 @SuppressWarnings("restriction")
-public class ProjectConfigSnapshot implements IProjectConfigSnapshot {
+public class ProjectConfigSnapshot {
 
 	private final String name;
 	private final URI path;
 	private final ImmutableSet<String> dependencies;
-	private final ImmutableList<ISourceFolderSnapshot> sourceFolders;
+	private final ImmutableSet<SourceFolderSnapshot> sourceFolders;
 
 	/** See {@link ProjectConfigSnapshot}. */
 	public ProjectConfigSnapshot(XIProjectConfig project) {
 		this(project.getName(), project.getPath(), project.getDependencies(),
-				project.getSourceFolders().stream().map(SourceFolderSnapshot::new).collect(Collectors.toList()));
+				project.getSourceFolders().stream().map(SourceFolderSnapshot::new)
+						.collect(ImmutableSet.toImmutableSet()));
 	}
 
 	/** See {@link ProjectConfigSnapshot}. */
 	public ProjectConfigSnapshot(String name, URI path, Iterable<String> dependencies,
-			Iterable<? extends ISourceFolderSnapshot> sourceFolders) {
+			Iterable<? extends SourceFolderSnapshot> sourceFolders) {
 		this.name = name;
 		this.path = path;
 		this.dependencies = ImmutableSet.copyOf(dependencies);
-		this.sourceFolders = ImmutableList.copyOf(sourceFolders);
+		this.sourceFolders = ImmutableSet.copyOf(sourceFolders);
 	}
 
-	@Override
+	/**
+	 * Return the project name.
+	 */
 	public String getName() {
 		return name;
 	}
 
-	@Override
+	/**
+	 * Return the project path.
+	 */
 	public URI getPath() {
 		return path;
 	}
 
-	@Override
+	/**
+	 * Return the project dependencies.
+	 */
 	public ImmutableSet<String> getDependencies() {
 		return dependencies;
 	}
 
-	@Override
-	public ImmutableCollection<? extends ISourceFolderSnapshot> getSourceFolders() {
+	/**
+	 * Return the project source folders.
+	 */
+	public ImmutableSet<? extends SourceFolderSnapshot> getSourceFolders() {
```

Comment:
Why the change from collection to set? Since, SourceFolderSnapshot does not implement `#equals()`, a set does not provide any benefit (and might actually be misleading).

Reply:
Added canonical methods to each `*Snapshot`.